crosscluster/logical: TestRandomStream failed

crosscluster/logical.TestRandomStream [failed](https://mesolite.cluster.engflow.com/invocations/default/69bae091-3d71-4fb7-9abd-b97c664fc33c?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL2xvZ2ljYWw6bG9naWNhbF90ZXN0) on master @ [9e83fa37459541594a9bb499f7cc18c155e37538](https://github.com/cockroachdb/cockroach/commits/9e83fa37459541594a9bb499f7cc18c155e37538):


```
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 24424
        	pkg/util/stop/stopper.go:436 +0x24f
        
        goroutine 24437 [chan receive, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*Scheduler).Start.func2({0xd0f6fd0, 0xc006ae7040})
        	pkg/kv/kvserver/rangefeed/scheduler.go:262 +0x65
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xd0f6fd0, 0xc006ae7040})
        	pkg/util/stop/stopper.go:438 +0x116
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 24424
        	pkg/util/stop/stopper.go:436 +0x24f
        
        goroutine 24442 [select, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/replica_rac2.(*streamCloseScheduler).run(0xc001fd35e0, {0xd099e40?, 0xc00592cea8?}, 0xc0061630e0)
        	pkg/kv/kvserver/kvflowcontrol/replica_rac2/close_scheduler.go:103 +0x2ad
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvflowcontrol/replica_rac2.(*streamCloseScheduler).Start.func1({0xd0f6fd0, 0xc006ae7040})
        	pkg/kv/kvserver/kvflowcontrol/replica_rac2/close_scheduler.go:53 +0x4d
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xd0f6fd0, 0xc006ae7040})
        	pkg/util/stop/stopper.go:438 +0x116
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 24424
        	pkg/util/stop/stopper.go:436 +0x24f
        
        goroutine 24009 [select]:
        github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs.func1()
        	pkg/util/admission/work_queue.go:479 +0x1cd
        created by github.com/cockroachdb/cockroach/pkg/util/admission.(*WorkQueue).startClosingEpochs in goroutine 23971
        	pkg/util/admission/work_queue.go:450 +0x8e
        
        goroutine 27905 [select, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func1()
        	pkg/sql/pgwire/server.go:1151 +0xe9
        created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 27904
        	pkg/sql/pgwire/server.go:1150 +0x439
        
        goroutine 24438 [select]:
        github.com/cockroachdb/cockroach/pkg/internal/client/requestbatcher.(*RequestBatcher).run(0xc004d70dc0, {0xd0f6fd0, 0xc006ae6fa0})
        	pkg/internal/client/requestbatcher/batcher.go:583 +0x850
        github.com/cockroachdb/cockroach/pkg/internal/client/requestbatcher.New.func1({0xd0f6fd0, 0xc006ae6fa0})
        	pkg/internal/client/requestbatcher/batcher.go:253 +0x116
        created by github.com/cockroachdb/cockroach/pkg/internal/client/requestbatcher.New in goroutine 24424
        	pkg/internal/client/requestbatcher/batcher.go:251 +0x70d
        
        goroutine 24008 [select]:
        github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
        	pkg/util/admission/work_queue.go:402 +0xbd
        created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 23971
        	pkg/util/admission/work_queue.go:399 +0x676
    logical_replication_job_test.go:1011: timed out waiting for events
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomStream1671865720
--- FAIL: TestRandomStream (237.86s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomStream.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

crosscluster/logical.TestRandomStream [failed](https://mesolite.cluster.engflow.com/invocations/default/3bd0beb6-708e-44bf-b0b3-0d2eae0c4b20?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL2xvZ2ljYWw6bG9naWNhbF90ZXN0) on master @ [91b3eee0f04549bf73bf1d35cc2da8a258ff5324](https://github.com/cockroachdb/cockroach/commits/91b3eee0f04549bf73bf1d35cc2da8a258ff5324):


```
        io.ReadFull(...)
        	GOROOT/src/io/io.go:354
        golang.org/x/net/http2.readFrameHeader({0xc00336f380, 0x9, 0x9}, {0xd0b8840, 0xc0044003c0})
        	external/org_golang_x_net/http2/frame.go:242 +0x9a
        golang.org/x/net/http2.(*Framer).ReadFrame(0xc00336f340)
        	external/org_golang_x_net/http2/frame.go:506 +0xf5
        google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams(0xc0073b5860, 0xc005741170, 0xbe38328)
        	external/org_golang_google_grpc/internal/transport/http2_server.go:637 +0x166
        google.golang.org/grpc.(*Server).serveStreams(0xc003a7e000, {0xd1805e0, 0xc0073b5860})
        	external/org_golang_google_grpc/server.go:963 +0x349
        google.golang.org/grpc.(*Server).handleRawConn.func1()
        	external/org_golang_google_grpc/server.go:905 +0x59
        created by google.golang.org/grpc.(*Server).handleRawConn in goroutine 16025
        	external/org_golang_google_grpc/server.go:904 +0x245
        
        goroutine 13563 [select]:
        github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue.func2()
        	pkg/util/admission/work_queue.go:402 +0xbd
        created by github.com/cockroachdb/cockroach/pkg/util/admission.initWorkQueue in goroutine 13548
        	pkg/util/admission/work_queue.go:399 +0x676
        
        goroutine 14058 [select, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*baseQueue).processLoop.func2({0xd11ef38, 0xc0021214d0})
        	pkg/kv/kvserver/queue.go:859 +0x1ae
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xd11ef38, 0xc0021214d0})
        	pkg/util/stop/stopper.go:438 +0x116
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 14026
        	pkg/util/stop/stopper.go:436 +0x24f
        
        goroutine 15866 [select]:
        github.com/cockroachdb/cockroach/pkg/sql/stats.(*Refresher).Start.func1({0xd1210b0, 0xc0057aa960})
        	pkg/sql/stats/automatic_stats.go:580 +0x44c
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xd1210b0, 0xc0057aa960})
        	pkg/util/stop/stopper.go:438 +0x116
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 13548
        	pkg/util/stop/stopper.go:436 +0x24f
        
        goroutine 15888 [select]:
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/load.(*runtimeLoadMonitor).run(0xc00868e280, {0xd1210b0, 0xc005034780})
        	pkg/kv/kvserver/load/node_capacity_provider.go:182 +0x30c
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/load.(*NodeCapacityProvider).Run.func1({0xd1210b0, 0xc005034780})
        	pkg/kv/kvserver/load/node_capacity_provider.go:77 +0x56
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xd1210b0, 0xc005034780})
        	pkg/util/stop/stopper.go:438 +0x116
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 13548
        	pkg/util/stop/stopper.go:436 +0x24f
    logical_replication_job_test.go:1011: timed out waiting for events
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomStream2579647182
--- FAIL: TestRandomStream (230.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomStream.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

crosscluster/logical.TestRandomStream [failed](https://mesolite.cluster.engflow.com/invocations/default/8929978a-9b03-4438-87ff-83c40cf677b4?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL2xvZ2ljYWw6bG9naWNhbF90ZXN0) on master @ [249793c9cfc6b75927639c33b8aed8d4b0e44ccb](https://github.com/cockroachdb/cockroach/commits/249793c9cfc6b75927639c33b8aed8d4b0e44ccb):


```
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc0017e8a00?, {0xd1b2180?, 0xc0077d8050?}, 0xc006988420?})
        	pkg/util/ctxgroup/ctxgroup.go:147 +0x85
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GoAndWait({0xd1b21b8, 0xc002105890}, {0xc0036b3b70, 0x2, 0xae865c0?})
        	pkg/util/ctxgroup/ctxgroup.go:223 +0x1bc
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).run(0xc00341a540, {0xd1b21b8, 0xc002105890}, {0xd211d00, 0xc0085c79a0}, 0x0)
        	pkg/kv/kvclient/rangefeed/rangefeed.go:394 +0x1431
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed.(*RangeFeed).start.func1({0xd1b2180, 0xc0085c79f0})
        	pkg/kv/kvclient/rangefeed/rangefeed.go:254 +0x73c
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0xd1b2180, 0xc0085c79f0})
        	pkg/util/stop/stopper.go:438 +0x116
        created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 12271
        	pkg/util/stop/stopper.go:436 +0x24f
        
        goroutine 13373 [select]:
        github.com/cockroachdb/cockroach/pkg/rpc.(*pipe).recv(0xc00265efd0, {0xd1b2180, 0xc008276780})
        	pkg/rpc/context.go:1216 +0x16d
        github.com/cockroachdb/cockroach/pkg/rpc.pipeReader.recv({0xc00265efd0?, 0xbebd6e8?}, {0xd1b2180, 0xc008276780}, {0xb785980, 0xc00e40def0})
        	pkg/rpc/context.go:1252 +0x59
        github.com/cockroachdb/cockroach/pkg/rpc.clientStream.RecvMsg(...)
        	pkg/rpc/context.go:1300
        github.com/cockroachdb/cockroach/pkg/rpc.muxRangeFeedClientAdapter.Recv({{0xd1ed430?, 0xc0056bcba0?}, 0xc00265f000?, 0xc006c72520?})
        	pkg/rpc/context.go:1135 +0x6b
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).receiveEventsFromNode(0xc00596b040, {0xd1b2180, 0xc008276780}, {0x7fc16072df48, 0xc006c72560}, 0xc0038c3380)
        	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:458 +0x70
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).startNodeMuxRangeFeed(0xc00596b040, {0xd1b2180, 0xc008276780}, {0xd170bf8, 0xc0082f54a0}, 0x1, 0xc0069ae970)
        	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:418 +0x8bd
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).establishMuxConnection.func1({0xd1b2180, 0xc008276730})
        	pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:349 +0x85
        github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*rangefeedMuxer).establishMuxConnection.Group.GoCtx.func2()
        	pkg/util/ctxgroup/ctxgroup.go:199 +0x11b
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	external/org_golang_x_sync/errgroup/errgroup.go:93 +0x92
        created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 13372
        	external/org_golang_x_sync/errgroup/errgroup.go:78 +0x125
        
        goroutine 13465 [select, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*testServer).Activate.func2()
        	pkg/server/testserver.go:898 +0x1b7
        created by github.com/cockroachdb/cockroach/pkg/server.(*testServer).Activate in goroutine 10922
        	pkg/server/testserver.go:895 +0x2eb
        
        goroutine 13618 [select, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func1()
        	pkg/sql/pgwire/server.go:1151 +0xe9
        created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 13611
        	pkg/sql/pgwire/server.go:1150 +0x439
    logical_replication_job_test.go:1012: timed out waiting for events
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestRandomStream1002524196
--- FAIL: TestRandomStream (230.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomStream.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/crosscluster/logical/BUILD.bazel
+++ b/pkg/crosscluster/logical/BUILD.bazel
@@ -34,7 +34,6 @@ go_library(
         "//pkg/ccl/changefeedccl/changefeedbase",
         "//pkg/ccl/utilccl",
         "//pkg/cloud",
-        "//pkg/cloud/externalconn",
         "//pkg/clusterversion",
         "//pkg/crosscluster",
         "//pkg/crosscluster/physical",

--- a/pkg/crosscluster/logical/logical_replication_job.go
+++ b/pkg/crosscluster/logical/logical_replication_job.go
@@ -8,11 +8,9 @@ package logical
 import (
 	"context"
 	"fmt"
-	"net/url"
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
-	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn"
 	"github.com/cockroachdb/cockroach/pkg/crosscluster"
 	"github.com/cockroachdb/cockroach/pkg/crosscluster/physical"
 	"github.com/cockroachdb/cockroach/pkg/crosscluster/replicationutils"
@@ -1092,21 +1090,17 @@ func getRetryPolicy(knobs *sql.StreamingTestingKnobs) retry.Options {
 func resolveDest(
 	ctx context.Context, execCfg *sql.ExecutorConfig, sourceURI string,
 ) (string, error) {
-	u, err := url.Parse(sourceURI)
+	configUri, err := streamclient.ParseConfigUri(sourceURI)
 	if err != nil {
 		return "", err
 	}
 
-	resolved := ""
-	err = execCfg.InternalDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
-		conn, err := externalconn.LoadExternalConnection(ctx, u.Host, txn)
-		if err != nil {
-			return err
-		}
-		resolved = conn.UnredactedConnectionStatement()
-		return nil
-	})
-	return resolved, err
+	clusterUri, err := configUri.AsClusterUri(ctx, execCfg.InternalDB)
+	if err != nil {
+		return "", err
+	}
+
+	return clusterUri.Serialize(), nil
 }
 
 func reloadDest(ctx context.Context, id jobspb.JobID, execCfg *sql.ExecutorConfig) (string, error) {

